internal/webhooks/clusterclass.go

```diff
@@ -117,8 +117,8 @@ func (webhook *ClusterClass) ValidateDelete(ctx context.Context, obj runtime.Obj
 
 	if len(clusters) > 0 {
 		// TODO(killianmuldoon): Improve error here to include the names of some clusters using the clusterClass.
-		return apierrors.NewForbidden(clusterv1.GroupVersion.WithResource("ClusterClass").GroupResource(), clusterClass.Name,
-			fmt.Errorf("cannot be deleted. %d clusters still using the ClusterClass", len(clusters)))
+		// TODO: change to "clusterClass cannot be deleted because it is used by Clusters: ...."
+		return fmt.Errorf("cannot be deleted. %d clusters still using the ClusterClass", len(clusters))
```

Comment:
What is the reasoning behind sometimes returning api errors and sometimes regular errors?

Reply:
Currently, the UX for the user using kubectl is not ideal due to several levels of error nesting that each message gets, so I have tried to remove some of them. But feedback and ideas will be appreciated

Reply:
I think my main point is, does it lead to better error messages when we return an apierror in a webhook. If yes, we should use it, if not we shouldn't.

Reply:
Not sure what we do with this result. I think I would prefer always using an api error because when otherwise the output always contains the error twice this gets problematic with the way we want to collect as many errors as possible and return those at once.

What do you think?

Reply:
It should be fixed now (always returning API errors to avoid duplications)